src/factories/FormioUtils.js

```diff
@@ -3,8 +3,9 @@ var formioUtils = require('formio-utils');
 module.exports = function() {
   return {
     checkVisible: function(component, row, data) {
-      if (!formioUtils.checkCondition(component, row, data)) {
-        if (!component.hasOwnProperty('clearOnHide') || component.clearOnHide !== false) {
+      var visible = formioUtils.checkCondition(component, row, data);
+      if (!visible) {
+        if (component.hasOwnProperty('clearOnHide') && component.clearOnHide === true) {
```

Comment:
I think this is going to have side effects. Previous forms did not have anything set for 'clearOnHide' so it should still default to clearOnHide if the field does not have the value. I think this reverses the logic.